runtime/gc_base/StringTable.cpp

```diff
@@ -568,6 +604,10 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 			}
 		}
 	}
+	bool isLatin1 = isASCII;
+	if (compressStrings && J2SE_VERSION(vm) >= J2SE_V17) {
```

Comment:
This doesn't seem the right place for this check, it doesn't take isUnicode into account. Also I think the check for isLatin1String can be incorporated into the check for isASCII, to avoid scanning the input twice.

Reply:
i.e. we only need to decode UTF8 which isn't already determined to be ASCII.

Reply:
Moved the code in question, check `isASCII` first, then check `isASCIIorLatin1` if required to minimize the overhead for pre-JDK17 levels, also added the check for `isUnicode` case.